File: Binary_Search_Tree/javascript/tree.js
function Tree() {
  this.root = null;
}

Tree.prototype.traverse = function () {
  this.root.visit(this.root);
};

Tree.prototype.search = function (value) {
  return this.root.search(value);
};

Tree.prototype.addValue = function (n) {
  const node = new Node(n);
  if (this.root == null) {
    this.root = node;
    this.root.x = width / 2;
    this.root.y = 16;
  } else {
    this.root.addNode(node);
  }
};

File: Binary_Search_Tree/python/tree.py
from node import Node
from typing import Optional


class Tree:

    def __init__(self):
        self.root = None

    def add_node(self, value: int) -> None:
        node = Node(value)
        if self.root is None:
            self.root = node
            return

        self.root.add_node(node)

    def traverse(self) -> None:
        self.root.traverse()

    def search(self, value: int) -> Optional[Node]:
        found = self.root.search(value)

        if found is None:
            print(f'{value} not found in tree')
        else:
            print(f'Found node={found} with value {value}')
        return found

    def print_tree(self):
        self.root.print_chunk(0)
